src/main/java/adris/altoclef/Settings.java

```diff
@@ -189,104 +176,75 @@
     private boolean useButlerWhitelist = true;
 
     /**
-     *
-     *
-     * Servers have different messaging plugins that change the way messages are displayed.
-     * Rather than attempt to implement all of them and introduce a big security risk,
-     * you may define custom whisper formats that the butler will watch out for.
-     *
+     * Servers have different messaging plugins that change the way messages are displayed. Rather than attempt to implement all of them and
+     * introduce a big security risk, you may define custom whisper formats that the butler will watch out for.
+     * <p>
      * Within curly brackets are three special parts:
-     *
-     * {from}: Who the message was sent from
-     * {to}: Who the message was sent to, butler will ignore if this is not your username.
-     * {message}: The message.
-     *
-     *
-     * WARNING: The butler will only accept non-chat messages as commands, but don't make this too lenient,
-     *      else you may risk unauthorized control to the bot. Basically, make sure that only whispers can
-     *      create the following messages.
+     * <p>
+     * {from}: Who the message was sent from {to}: Who the message was sent to, butler will ignore if this is not your username. {message}:
+     * The message.
+     * <p>
+     * <p>
+     * WARNING: The butler will only accept non-chat messages as commands, but don't make this too lenient, else you may risk unauthorized
+     * control to the bot. Basically, make sure that only whispers can create the following messages.
      */
-    private String[] whisperFormats = new String[] {
+    private String[] whisperFormats = {
+            "\\[{from} -> {to}\\] {message}",
             "{from} whispers to you: {message}",
-            "{from} whispers: {message}",
-            "\\[{from} -> {to}\\] {message}"
-    };
+            "{from} whispers: {message}"
+    }; // TODO: 2021-05-22 a regex would be better
```

Comment:
Take a look at how it's implemented in [WhisperChecker.java](https://github.com/adrisj7-AltoClef/altoclef/blob/85af4358c93f0e5ad1575bc0b6a494626ffbb105/src/main/java/adris/altoclef/butler/WhisperChecker.java#L64).

I use regex to parse this, but this format is designed so users who aren't programmers can easily write these expressions, and don't have to learn about regex when it doesn't interest them.

Reply:
I mean, maybe use regexes and normal expressions. Also, it's hardcoded rn, so it's not really user configurable.